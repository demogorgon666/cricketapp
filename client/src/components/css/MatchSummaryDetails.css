@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');

body{
    font-family: 'Montserrat', sans-serif;
    color: #272727;
}

.top-container{
    
    display: grid;
    grid-template-columns: 60% 35%;
    grid-template-rows: 40% 40%;
    grid-template-areas: "summary matchdetails"
                        "playingXI matchdetails";
    grid-gap: 10px;
    justify-content: space-between; 
    height: auto;
    padding: 20px;
}

.top-container .top-left-container{
    display: grid;
    grid-template-rows: 10% 15% 25% 15% 25%;
    border: 1px solid #272727;
    border-radius: 5px
}

.top-container .teamone-top-players, .teamtwo-top-players{
    display: grid;
    grid-template-columns: auto 15% auto 15%;
    border-top: 1px solid #272727;   
    border-bottom: 1px solid #272727;   
    padding: 2px;
    grid-gap: 3px; 

}
.top-container .bottom-left-container{
    display: grid;
    grid-template-rows: 10% 90%;
    border: 1px solid #272727;
    border-radius: 5px;
}

.bottom-left-container .playingXI{
    display: grid;
    grid-template-columns: 50% 50%; 
    grid-template-rows: 12% 80%; 
    grid-gap: 5px;
}
.match_details{
    grid-area: matchdetails;
    height: fit-content;
    }

    
.top-title{
    font-weight: 500;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    background: #2980B9;
    color: white;
    padding: 5px;
    border-radius: 5px;
}

/* 
.topOfSummary{
    display: grid;
    grid-template-columns: 60% 10% 30%;
    
}

.scoreCardSummary{
    border: 1px solid #272727;
    height: auto;
    margin-left: 15px;
    display: grid;
    grid-template-rows: 10% 10% 30% 10% 30%
}

.matchDetails{
    border: 1px solid green;
    margin-right: 80px;
}

.title{
    background-color: #2980B9;
    
}

.players{
    display: grid;
    grid-template-columns: 50% 50%;
    
}

.playerLeft, .playerRight{
    border: 1px solid green;
    height: 150px;
    display: flex;
    flex-direction: rows;
    justify-content: space-between;
    font-size: 20px;
}
.playerinfo{
    margin-top: 10px;
    margin-left: 100px;
}
.playerScores{
    margin-top: 10px;
    margin-right: 100px;
}


.match-title{
    text-align: left;
    margin-top: 8px;
    margin-left: 8px;
}
.match-values{
    text-align: left;
    margin-top: 8px;
    margin-left: 8px;

}

.playing11{
    border: 1px solid blue;
    height: 500px;
    margin-left: 15px;
    display: grid;
    grid-template-rows: 10% 90%
}
.team-names{
    margin-left: 15px;
    display: grid;
    grid-template-columns: 50% 50%;
    border-bottom: 1px solid black;
    display: inline;
    
}
 .team-players-list, .team-names{
    text-align: left;
    margin-left: 15px;
    margin-top: 15px;
    display: grid;
    grid-template-columns: 50% 50%;
    border-bottom: 1px solid black; 
    
}

.team-names{
    border-bottom: 1px solid black;
}  */