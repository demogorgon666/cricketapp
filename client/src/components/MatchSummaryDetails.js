import React, { Component } from "react";
import { getmatchdetailbyId } from "../actions/Matches.js";
import { connect } from "react-redux";
import "./css/MatchSummaryDetails.css";
import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

export class MatchSummaryDetails extends Component {
  componentDidMount() {
    this.props.getmatchdetailbyId(this.props.match_id);
  }
  render() {
    return (
      <div>
        {this.props.isLoading ? (
          <div style={{ margin: "400px" }}>
            <Loader
              type="TailSpin"
              color="#2980b9"
              height="100"
              width="100"
            />
          </div>
        ) : (
            <div style={{ marginTop: 150 + "px" }}>
              {this.props.match.map(match => (
                <div className="top-container">
                  <div className="top-left-container">
                    <div className="top-title">Summary Scorecard</div>

                    <div style={{ margin: "15px 0 0 8px" }}>
                      {match.teamone_name[0].teamone_name}
                    </div>

                    <div className="teamone-top-players">
                      <div style={{ fontWeight: "500" }}>
                        {match.team1_batsman.map(t1_batsman => (
                          <div>{t1_batsman.player_name}</div>
                        ))}
                      </div>
                      <div
                        style={{
                          borderRight: "1px solid #272727"
                        }}
                      >
                        {match.team1_batsman.map(t1_batsman => (
                          <div>
                            {t1_batsman.total_runs} ({t1_batsman.total_ball})
                        </div>
                        ))}
                      </div>
                      <div style={{ fontWeight: "500" }}>
                        {match.team1_bowler.map(t1_bowler => (
                          <div>{t1_bowler.player_name}</div>
                        ))}
                      </div>
                      <div>
                        {match.team1_bowler.map(t1_bowler => (
                          <div>
                            {t1_bowler.total_wicket}/{t1_bowler.given_runs} (
                          {t1_bowler.overs})
                        </div>
                        ))}
                      </div>
                    </div>

                    <div style={{ margin: "15px 0 0 8px" }}>
                      {match.teamtwo_name[0].teamtwo_name}
                    </div>
                    <div className="teamtwo-top-players">
                      <div style={{ fontWeight: "500" }}>
                        {match.team2_batsman.map(t2_batsman => (
                          <div>{t2_batsman.player_name}</div>
                        ))}
                      </div>
                      <div
                        style={{
                          borderRight: "1px solid #272727"
                        }}
                      >
                        {match.team2_batsman.map(t2_batsman => (
                          <div>
                            {t2_batsman.total_runs} ({t2_batsman.total_ball})
                        </div>
                        ))}
                      </div>
                      <div style={{ fontWeight: "500" }}>
                        {match.team2_bowler.map(t2_bowler => (
                          <div>{t2_bowler.player_name}</div>
                        ))}
                      </div>
                      <div>
                        {match.team2_bowler.map(t2_bowler => (
                          <div>
                            {t2_bowler.total_wicket}/{t2_bowler.given_runs} (
                          {t2_bowler.overs})
                        </div>
                        ))}
                      </div>
                    </div>
                  </div>
                  <div
                    className="match_details"
                    style={{ border: "1px solid #272727", borderRadius: "5px" }}
                  >
                    <div className="top-title"> Match Details</div>
                    <div style={{ margin: "5px 0 0 8px" }}>Series</div>
                    <div style={{ fontWeight: "500", margin: "0 0 0 15px" }}>
                      {" "}
                      {match.match_details[0].competition}
                    </div>
                    <div style={{ margin: "3px 0 0 8px" }}> Date</div>

                    <div style={{ fontWeight: "500", margin: "0 0 0 15px" }}>
                      {match.date.map(date => (
                        <div>{date.match_date}</div>
                      ))}
                    </div>
                    <div style={{ margin: "3px 0 0 8px" }}> Toss</div>
                    <div style={{ fontWeight: "500", margin: "0 0 0 15px" }}>
                      {" "}
                      {match.match_details[0].toss_winner_team} elected to{" "}
                      {match.match_details[0].toss_decision}
                    </div>
                    <div style={{ margin: "3px 0 0 8px" }}> Venue</div>
                    <div style={{ fontWeight: "500", margin: "0 0 0 15px" }}>
                      {" "}
                      {match.match_details[0].venue_name},{" "}
                      {match.match_details[0].venue_city}
                    </div>

                    <div style={{ margin: "3px 0 0 8px" }}> Umpire</div>
                    {match.umpires.map(umpire => (
                      <div style={{ fontWeight: "500", margin: "0 0 0 15px" }}>
                        {" "}
                        {umpire.umpire_name}
                      </div>
                    ))}
                  </div>

                  <div
                    className="bottom-left-container"
                    style={{ borderRadius: "5px" }}
                  >
                    <div className="top-title"> Playing XI</div>
                    <div className="playingXI">
                      <div
                        style={{
                          borderBottom: "1px solid #272727",
                          margin: "15px 0 0 8px"
                        }}
                      >
                        {match.teamone_name.map(teamone => teamone.teamone_name)}
                      </div>
                      <div
                        style={{
                          borderBottom: "1px solid #272727",
                          margin: "15px 0 0 8px"
                        }}
                      >
                        {match.teamtwo_name.map(teamtwo => teamtwo.teamtwo_name)}
                      </div>
                      <div style={{ margin: "0 0 0 10px", fontWeight: "500" }}>
                        {match.team_one_XI.map(team1_XI => (
                          <div>{team1_XI.teamone_players} </div>
                        ))}
                      </div>
                      <div style={{ margin: "0 0 0 10px", fontWeight: "500" }}>
                        {match.team_two_XI.map(team2_XI => (
                          <div>{team2_XI.teamtwo_players} </div>
                        ))}
                      </div>
                    </div>
                  </div>
                </div>
              ))
              }
            </div>)}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  match: state.matchreducer.match,
  isLoading: state.LoadingReducer.isLoading
});

export default connect(
  mapStateToProps,
  { getmatchdetailbyId }
)(MatchSummaryDetails);
