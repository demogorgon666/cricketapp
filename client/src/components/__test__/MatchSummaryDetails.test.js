import React from "react";
import { shallow } from "enzyme";
import MatchSummaryDetails from "../MatchSummaryDetails"
import { Provider } from "react-redux";
import store from "../../store";


// const matchLandingPage = jest.fn();
const wrapper = shallow(
    <Provider store={store}>
        <MatchSummaryDetails />
    </Provider>
);


describe("Testing of MatchSummaryDetails Component", () => {

    it("should render the component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    // it("should have one h2 header", () => {
    //     expect(wrapper.find("h2").length).toBe(1);
    // });
    // it("should have one h2 content with 'Recent Matches' content", () => {
    //     expect(wrapper.find(".h2-recent-matches").text()).toBe("MS Dhoni");
    // });
    // it("should have one div tag with 'Player of the match", () => {
    //     expect(wrapper.find(".short-summary-with-player-of-the-match span").text()).toBe("Player of the match");
    // });



});
