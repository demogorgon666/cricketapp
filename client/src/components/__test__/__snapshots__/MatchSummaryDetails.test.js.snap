// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of MatchSummaryDetails Component should render the component 1`] = `
<div>
  <div
    style={
      Object {
        "marginTop": "210px",
      }
    }
  >
    <div
      className="top-container"
    >
      <div
        className="top-left-container"
      >
        <div
          className="top-title"
          id="summary-title"
        >
          Summary Scorecard
        </div>
        <div
          id="team-one-name-summary"
          style={
            Object {
              "padding": "10px",
            }
          }
        >
          Australia
        </div>
        <div
          className="teamone-top-players"
        >
          <div
            style={
              Object {
                "fontWeight": "500",
                "margin": "10px",
              }
            }
          >
            <div
              id="team-one-tbm"
            >
              PJ Hughes
            </div>
          </div>
          <div
            style={
              Object {
                "borderRight": "1px solid #272727",
                "padding": "10px",
              }
            }
          >
            <div
              id="team-one-tbm-runs"
            >
              115
               (
              151
              )
            </div>
          </div>
          <div
            style={
              Object {
                "fontWeight": "500",
                "margin": "10px",
              }
            }
          >
            <div
              id="team-one-tbn"
            >
              DW Steyn
            </div>
          </div>
          <div
            style={
              Object {
                "margin": "10px",
              }
            }
          >
            <div
              id="team-one-tbn-wickets"
            >
              3
              /
              88
               (
              25
              )
            </div>
          </div>
        </div>
        <div
          id="team-two-name"
          style={
            Object {
              "padding": "10px",
            }
          }
        >
          South Africa
        </div>
        <div
          className="teamtwo-top-players"
        >
          <div
            style={
              Object {
                "fontWeight": "500",
                "margin": "10px",
              }
            }
          >
            <div
              id="team-two-tbn"
            >
              JP Duminy
            </div>
          </div>
          <div
            style={
              Object {
                "borderRight": "1px solid #272727",
                "padding": "10px",
              }
            }
          >
            <div
              id="team-two-btm-score"
            >
              73
               (
              152
              )
            </div>
          </div>
          <div
            style={
              Object {
                "fontWeight": "500",
                "margin": "10px",
              }
            }
          >
            <div
              id="team-two-top-bowler-name"
            >
              MG Johnson
            </div>
          </div>
          <div
            style={
              Object {
                "margin": "10px",
              }
            }
          >
            <div
              id="team-two-top-bowler-wicket"
            >
              4
              /
              51
               (
              15
              )
            </div>
          </div>
        </div>
      </div>
      <div
        className="match_details"
        style={
          Object {
            "borderRadius": "5px",
          }
        }
      >
        <div
          className="top-title"
          id="match-conatiner-title"
        >
          Match Details
        </div>
        <div
          id="competition-name-text"
          style={
            Object {
              "margin": "5px 0 0 8px",
            }
          }
        >
          Series
        </div>
        <div
          id="competition-name"
          style={
            Object {
              "fontWeight": "500",
              "margin": "0 0 0 15px",
            }
          }
        >
          others
        </div>
        <div
          id="match-date-text"
          style={
            Object {
              "margin": "3px 0 0 8px",
            }
          }
        >
          Date
        </div>
        <div
          style={
            Object {
              "fontWeight": "500",
              "margin": "0 0 0 15px",
            }
          }
        >
          <div
            id="match-date-value"
          >
            Mar 07, 2009
          </div>
        </div>
        <div
          id="match-toss-text"
          style={
            Object {
              "margin": "3px 0 0 8px",
            }
          }
        >
          Toss
        </div>
        <div
          id="toss-and-decision"
          style={
            Object {
              "fontWeight": "500",
              "margin": "0 0 0 15px",
            }
          }
        >
          Australia
           elected to
           
          bat
        </div>
        <div
          id="match-venue-text"
          style={
            Object {
              "margin": "3px 0 0 8px",
            }
          }
        >
          Venue
        </div>
        <div
          id="match-venue-value"
          style={
            Object {
              "fontWeight": "500",
              "margin": "0 0 0 15px",
            }
          }
        >
          Kingsmead
          ,
           
          Durban
        </div>
        <div
          id="match-umpire-text"
          style={
            Object {
              "margin": "3px 0 0 8px",
            }
          }
        >
          Umpire
        </div>
        <div
          id="match-umpire-name"
          style={
            Object {
              "fontWeight": "500",
              "margin": "0 0 0 15px",
            }
          }
        >
          Asad Rauf
        </div>
      </div>
      <div
        className="bottom-left-container"
        style={
          Object {
            "borderRadius": "5px",
          }
        }
      >
        <div
          className="top-title"
          id="playing-XI-title"
        >
          Playing XI
        </div>
        <div
          className="playingXI"
        >
          <div
            id="team-one-p-name"
            style={
              Object {
                "borderBottom": "1px solid #272727",
                "margin": "15px 0 0 8px",
              }
            }
          >
            Australia
          </div>
          <div
            id="team-two-p-name"
            style={
              Object {
                "borderBottom": "1px solid #272727",
                "margin": "15px 0 0 8px",
              }
            }
          >
            South Africa
          </div>
          <div
            style={
              Object {
                "fontWeight": "500",
                "margin": "0 0 0 10px",
              }
            }
          >
            <div
              id="team-one-playingXI"
            >
              MJ Clarke
            </div>
          </div>
          <div
            style={
              Object {
                "fontWeight": "500",
                "margin": "0 0 0 10px",
              }
            }
          >
            <div
              id="team-two-playingXI"
            >
              M Ntini
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
